packages/price-reporter/src/client.ts

```diff
@@ -0,0 +1,67 @@
+import { CHAIN_IDS, type ChainId, getSDKConfig } from "@renegade-fi/core";
+import { getDefaultQuoteToken } from "@renegade-fi/token";
+import type { AxiosRequestConfig } from "axios";
+import axios from "axios";
+import { ResultAsync, okAsync } from "neverthrow";
+import { PRICE_REPORTER_ROUTE } from "./constants.js";
+import { HttpError, type PriceReporterError } from "./error.js";
+
+export class PriceReporterClient {
+    constructor(private readonly baseUrl: string) {}
+
+    static new(chainId: ChainId) {
+        const config = getSDKConfig(chainId);
+        return new PriceReporterClient(`https://${config.priceReporterUrl}:3000`);
+    }
+
+    static newArbitrumMainnetClient() {
+        const config = getSDKConfig(CHAIN_IDS.ArbitrumMainnet);
+        return new PriceReporterClient(`https://${config.priceReporterUrl}:3000`);
+    }
+
+    static newArbitrumSepoliaClient() {
+        const config = getSDKConfig(CHAIN_IDS.ArbitrumSepolia);
+        return new PriceReporterClient(`https://${config.priceReporterUrl}:3000`);
+    }
+
+    /**
+     * Get the price of a token from a given exchange
+     */
+    public getPrice(
+        exchange: string,
+        address: `0x${string}`,
+        quote: `0x${string}`,
+    ): ResultAsync<number, PriceReporterError> {
+        const route = PRICE_REPORTER_ROUTE(exchange, address, quote);
+        return this.get<string>(route).andThen((textPrice) => {
+            const price = Number(textPrice);
```

Comment:
Nit: I would use a `fromThrowable` wrapper around `parseFloat` here, since we're already using `neverthrow`